File: Bug_Memory/Bug_Memory/Resources/CocosBuilder/Published-iOS/MainScene.js
//
// MainScene class
//
var MainScene = function(){};

// Create callback for button
MainScene.prototype.onCleanTextureCachePressed = function()
{	
    cc.log('Remove');
    cc.SpriteFrameCache.getInstance().removeUnusedSpriteFrames();
    cc.TextureCache.getInstance().removeUnusedTextures();
};

MainScene.prototype.onDumpTextureInfoPressed = function()
{	
    cc.log('Dump');
    cc.TextureCache.getInstance().dumpCachedTextureInfo();
};

MainScene.prototype.onBackPressed = function()
{	
    cc.log('Back');
};

MainScene.prototype.onNextPressed = function()
{	
    cc.log('Next');
    var scene = cc.BuilderReader.loadAsScene("SecondScene");
    cc.Director.getInstance().replaceScene(scene);
};

File: Bug_Memory/Bug_Memory_PureObjC/MainScene.m
//
//  MainScene.m
//  Bug_Memory_PureObjC
//
//  Created by Manel Mateos Ramírez on 21/01/13.
//  Copyright 2013 Divertap. All rights reserved.
//

#import "MainScene.h"
#import "SecondScene.h"
#import "CCBReader.h"

@implementation MainScene

- (void) didLoadFromCCB {
    CCLOG(@"%@.didLoadFromCCB",NSStringFromClass([self class]));
}

-(void) onCleanTextureCachePressed: (id) sender
{
    CCLOG(@"Remove");
    [[CCSpriteFrameCache sharedSpriteFrameCache] removeUnusedSpriteFrames];
    [[CCTextureCache sharedTextureCache] removeUnusedTextures];
};

-(void) onDumpTextureInfoPressed: (id) sender
{
    CCLOG(@"Dump");
    [[CCTextureCache sharedTextureCache] dumpCachedTextureInfo];
};

-(void) onBackPressed: (id) sender
{
    CCLOG(@"Back");

};

-(void) onNextPressed: (id) sender
{
    CCLOG(@"Next");
    CCScene *scene = [CCBReader sceneWithNodeGraphFromFile:@"SecondScene.ccbi"];
    [[CCDirector sharedDirector] replaceScene: scene];
}

-(void) onDump: (id) sender {
    CCLOG(@"Hola");
}

@end
